pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,254 @@
+	if err != nil {
+		return false, err
+	}
+
+	return accountIsmemberofOrg, nil
+}
+
+// Validate cache and refresh if its stale
+func (o *orgValidator) validateCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (bool, error) {
+
+	isStale, err := o.checkIfOrgAccountListIsStale(ctx)
+	if err != nil {
+		return isStale, err
+	}
+
+	// refresh the account map
+	if isStale {
+		_, err = o.reloadAccountList(ctx, orgClient, false)
+		if err != nil {
+			return isStale, err
+		}
+	}
+	return isStale, nil
+}
+
+func (o *orgValidator) lookupCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string, reValidatedcache bool) (bool, error) {
+
+	o.mutex.RLock()
+	orgAccountList := o.orgListAccountMap
+	o.mutex.RUnlock()
+
+	_, accoutIsmemberOfOrg := orgAccountList[accoundIDofNode]
+
+	// Retry if it doesn't exist in cache and cache was not revalidated
+	if !accoutIsmemberOfOrg && !reValidatedcache {
+		orgAccountList, err := o.refreshCache(ctx, orgClient)
+		if err != nil {
+			return false, err
+		}
+		_, accoutIsmemberOfOrg = orgAccountList[accoundIDofNode]
+	}
+
+	return accoutIsmemberOfOrg, nil
+}
+
+// If cache miss happens, refresh list with new cache and check if element exist
+func (o *orgValidator) refreshCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (map[string]bool, error) {
+	remTries := o.getRetries()
+
+	orgAccountList := make(map[string]bool)
+	if remTries <= 0 {
+		return orgAccountList, nil
+	}
+
+	orgAccountList, err := o.reloadAccountList(ctx, orgClient, true)
+	if err != nil {
+		return nil, err
+	}
+
+	o.decrRetries()
+
+	return orgAccountList, nil
+}
+
+// Check if the org account list is stale.
+func (o *orgValidator) checkIfOrgAccountListIsStale(ctx context.Context) (bool, error) {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+
+	// Map is empty that means this is first time plugin is being initialised
+	if len(o.orgListAccountMap) == 0 {
+		return true, nil
+	}
+
+	// Get the timestamp from config
+	existingTimestamp := o.orgListAccountMapCreationTime
+
+	currTimeStamp := time.Now().UTC()
+
+	// Check diff of timestamp of org acc map & current time if its more than ttl, refresh the list
+	if currTimeStamp.Sub(existingTimestamp) >= time.Duration(o.orgAccountListCacheTTL) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// Get the list of accounts belonging to organization and catch them
+func (o *orgValidator) reloadAccountList(ctx context.Context, orgClient organizations.ListAccountsAPIClient, catchBurst bool) (map[string]bool, error) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	// Make sure : map is not updated from different go routine.
+	// * if we make this call before ttl expires, we are doing retry/catchburst that means we need to bypass this validation
+	if !catchBurst && len(o.orgListAccountMap) != 0 && time.Now().UTC().Sub(o.orgListAccountMapCreationTime) <= time.Duration(o.orgAccountListCacheTTL) {
```

Comment:
time.Now().UTC().Sub(o.orgListAccountMapCreationTime) <= time.Duration(o.orgAccountListCacheTTL is used in multiple places, may we move to a function?

and can you add some unit tests for this code?

Reply:
Done moved to function. And added tests for it. As this is private function, let me know if its not acceptable in the way I wrote tests for this as well as other functions in this file.